packages/astro/src/server/middleware.ts

```diff
@@ -162,7 +162,7 @@ async function instrumentRequest(
         const newResponseStream = new ReadableStream({
           start: async controller => {
             for await (const chunk of originalBody) {
-              const html = typeof chunk === 'string' ? chunk : decoder.decode(chunk);
+              const html = typeof chunk === 'string' ? chunk : decoder.decode(chunk, { stream: true });
```

Comment:
If we are making this change, we shouldn't reuse the TextDecoder, since the shared memory can contain the previous chunk information. Also, maybe we should add `fatal: true`?

Reply:
Don't we need to have the previous chunk information to be able to properly stick the chunks back together when we hit an "within character" chunk boundary? I am not familiar with this API at all so I am kinda making blind assumptions here.

Reply:
Yes. I misread the decoder declaration location. I thought it was in the main scope, but it actually has the lifespan of the `newResponseStream` which is indeed correct.